src/poetry/utils/password_manager.py

```diff
@@ -195,18 +195,19 @@ def delete_pypi_token(self, repo_name: str) -> None:
     def get_http_auth(self, repo_name: str) -> dict[str, str | None] | None:
         username = self._config.get(f"http-basic.{repo_name}.username")
         password = self._config.get(f"http-basic.{repo_name}.password")
-        if not username and not password:
+
+        if not username:
             return None
 
         if not password:
             if self.use_keyring:
                 password = self.keyring.get_password(repo_name, username)
-            else:
+            elif not username:
                 return None
```

Comment:
previous comment was "resolved" too soon and I do not have power to unresolve it, so leaving another!

this code is indeed unhittable.

or if I am wrong: please write a unit test covering it

Reply:
@dimbleby I thought I responded with the case. The test case already exists https://github.com/python-poetry/poetry/pull/9850/files#diff-9290d504b4b225fb1efa7f54d44cc4ed335659c90d0c6ca558176d326b3d1355R138.

You can switch that to an `else` statement and run the test case, it will fail.

Reply:
the `return None`  is not hit by your testcase, as you easily can verify.  Just remove both lines of code.

Reply:
Fair enough. I can drop both lines :smile:

Reply:
Dropped and simplified the if block now that I follow you.